src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -755,7 +755,16 @@ private bool TryTakeWithNoTimeValidation([MaybeNullWhen(false)] out T item, int
                     }
                 }
             }
+
+            if (waitForSemaphoreWasSuccessful)
+            {
+                Debug.Assert(item != null);
```

Comment:
I'm not sure if this assertion is actually correct. Is it possible to have - say - `BlockingCollection<string?>`? If that's legal, we should remove this assert.

Reply:
> Is it possible to have - say - `BlockingCollection<string?>`?

Yes, that's valid.  Please revert the added code.  It's a little disheartening this didn't cause any existing tests to fail.  Sigh.

Reply:
I'll remove the assertion and add an explicit unit test.